### PR TITLE
SR reconstructions of LR labelmaps

### PR DESCRIPTION
**Pull request by pdedumast**

### CODE REVIEW

**Comment by t-sanchez** [comment]
When the code is run with the configuration file
```
{
    "04": [
        {
            "sr-id": 2,
            "stacks": [1,4,7],
            "custom_interfaces": {
                "skip_svr": true,
                "do_nlm_denoising":true,
                "do_reconstruct_labels": true
            }
        }
    ]
}
```

it fails with the following error during the SDI computation:
```
Traceback (most recent call last):^M
  File "/opt/conda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 398, in run^M
    runtime = self._run_interface(runtime)^M
  File "/opt/conda/lib/python3.7/site-packages/pymialsrtk/interfaces/reconstruction.py", line 556, in _run_interface^M
    self.inputs.stacks_order)^M
  File "/opt/conda/lib/python3.7/site-packages/pymialsrtk/interfaces/utils.py", line 104, in reorder_by_run_ids^M
    id_and_files_ordered.append([[s, f[1]] for f in orig_order if ('run-' + str(s)) in f[1]][0])^M
IndexError: list index out of range^
```

File: pymialsrtk/workflows/postproc_stage.py [hunk 1]
```diff
@@ -33,13 +30,16 @@ def convert_ga(ga):
...
+        p_do_reconstruct_labels=False,
         name="postproc_stage"
 ):
     """Create a SR preprocessing workflow
     Parameters
     ----------
     ::
         name : name of workflow (default: preproc_stage)
+        p_ga
+        p_do_anat_orientation
```
Comment (t-sanchez) [inline pymialsrtk/workflows/postproc_stage.py:42]: Missing `p_do_reconstruct_labels`?

**Review by t-sanchez — CHANGES_REQUESTED** [review]

File: pymialsrtk/interfaces/preprocess.py [hunk 2]
```diff
@@ -1564,14 +1572,33 @@ def _crop_image_and_mask(self,
...
+            writer.Execute(label_cropped)
 
     def _run_interface(self, runtime):
 
         try:
             self._crop_image_and_mask(
                 self.inputs.input_image,
-                self.inputs.input_mask
+                self.inputs.input_mask,
+                self.inputs.input_label
```
[the remark this answers is not in the document]
Comment (pdedumast) [reply pymialsrtk/interfaces/preprocess.py:1601]: Yes, when input_label is not connected, `self.inputs.input_label=None`.

**Comment by pdedumast** [comment]
@t-sanchez , I fixed the processing with `"do_nlm_denoising":true`. Regarding the `mask_hr_label` module, I think you were not on the very last commit, though it is in the postprocessing stage atm.

File: pymialsrtk/interfaces/reconstruction.py [hunk 3]
```diff
@@ -475,13 +475,14 @@ class MialsrtkSDIComputationInputSpec(BaseInterfaceInputSpec):
...
                                     'specify the order of the stacks')
+    label_id = traits.Int(-1, mandatory=False, usedefault=True)
 
 
 class MialsrtkSDIComputationOutputSpec(TraitedSpec):
     """Class used to represent outputs of the
     MialsrtkSDIComputation interface."""
 
-    output_sdi = File(desc='Output scattered data interpolation image file')
+    output_hr = File(desc='Output scattered data interpolation image file')
```
Comment (t-sanchez) [inline pymialsrtk/interfaces/reconstruction.py:485]: Maybe a small thing, but why was it renamed?
It seems that we now have a mix of `output_hr` and `input_sdi` below.

File: pymialsrtk/workflows/recon_stage.py [hunk 4]
```diff
@@ -155,7 +170,7 @@ def create_recon_stage(p_paramTV,
         recon_stage.connect(srtkImageReconstruction, "output_sdi",
                             sdiComputation, "input_reference")
 
-        recon_stage.connect(sdiComputation, "output_sdi",
+        recon_stage.connect(sdiComputation, "output_hr",
```
Comment (t-sanchez) [inline pymialsrtk/workflows/recon_stage.py:173]: Inconsistent usage of `output_hr` and `input_sdi`. This makes it confusing IMO.

File: pymialsrtk/workflows/recon_stage.py [hunk 5]
```diff
@@ -189,7 +221,7 @@ def create_recon_stage(p_paramTV,
                             srtkHRMask, "input_image")
 
     if p_do_nlm_denoising:
-        recon_stage.connect(sdiComputation, "output_sdi",
+        recon_stage.connect(sdiComputation, "output_hr",
```
Comment (t-sanchez) [inline pymialsrtk/workflows/recon_stage.py:224]: Same as above: `output_hr` goes into `output_sdi`.

**Review by t-sanchez — CHANGES_REQUESTED** [review]

**Pull request merged**